stdlib/LinearAlgebra/src/blas.jl

```diff
@@ -1545,6 +1545,9 @@ for (mfname, elty) in ((:dsymm_,:Float64),
             if size(B,2) != n
                 throw(DimensionMismatch(lazy"B has second dimension $(size(B,2)) but needs to match second dimension of C, $n"))
             end
+            if j != size(B,1)
```

Comment:
It's more complicated. We need to take into account whether the symmetric factor is supposed to be on the left or on the right. I think I have a fix for the size checks. The question is, how to get the error messages right because of the dependence on `side`.

Reply:
Yes, we haven't had rectangular tests (apparently), so I'm going to add some.